src/AddProjectsPage/index.tsx

```diff
@@ -318,58 +308,67 @@ class AddProjectsPage extends React.Component<
     }
   };
 
-  public teamFilter = () => {
+  teamFilter = () => {
     this.filter('teamSearch', 'team');
   };
 
-  public tagFilter = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+  tagFilter = (e: React.KeyboardEvent<HTMLInputElement>): void => {
     e.persist();
     this.filter('tagSearch', 'tags');
-    if (e.keyCode === 13) {
-      var inputValue = (document.getElementById(
-        'tagSearch'
-      )! as HTMLInputElement).value;
-      var array = this.state.tags!.slice();
-      if (array.indexOf(inputValue.toLowerCase()) === -1) {
-        array.push(inputValue.toLowerCase());
-      }
-      this.setState({ tags: array }, () => {
-        this.toggleDropdown(e, 'new-tags-dropdown');
-      });
+    const ENTER_KEY = 13;
+    // on 'enter', adds new tag to array in state if it doesnt already exist
+    if (e.keyCode === ENTER_KEY) {
+      var value = (document.getElementById('tagSearch')! as HTMLInputElement)
+        .value;
+      this.addValueToStateArray('tags', value);
+      this.toggleDropdown(e, 'new-tags-dropdown');
     }
   };
 
-  public categoryFilter = () => {
+  categoryFilter = () => {
     this.filter('categorySearch', 'category');
   };
 
-  public handleImageUpload = (e: React.FormEvent<HTMLButtonElement>): void => {
+  // opens initial file window to select images from local computer
+  // saves files to state
+  handleImageInitialBtn = (e: React.FormEvent<HTMLInputElement>): void => {
+    let files = e.currentTarget.files! as FileList;
+    this.setState({ files: files } as any);
+  };
+
+  // clears images from image preview section and state
+  handleImageClear = (e: React.FormEvent<HTMLButtonElement>): void => {
+    const preview = document.getElementById('new-project-image-preview')!;
+    while (preview.firstChild) {
+      preview.removeChild(preview.firstChild);
```

Comment:
We need to stop doing so many DOM manipulations ourselves and let React do the work.

Reply:
imagePreview is re-worked so it doesnt use DOM manipulations